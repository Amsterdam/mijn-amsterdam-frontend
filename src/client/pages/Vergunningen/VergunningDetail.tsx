import { AanbiedenDienstenContent } from './detail-page-content/AanbiedenDiensten';
import { ERVV } from './detail-page-content/ERVV';
import { EvenementMelding } from './detail-page-content/EvenementMelding';
import { EvenementVergunning } from './detail-page-content/EvenementVergunning';
import { Flyeren } from './detail-page-content/Flyeren';
import { Nachtwerkontheffing } from './detail-page-content/Nachtwerkontheffing';
import { Omzettingsvergunning } from './detail-page-content/Omzettingsvergunning';
import { RvvHeleStad } from './detail-page-content/RvvHeleStad';
import { RvvSloterweg } from './detail-page-content/RvvSloterweg';
import { TVMRVVObject } from './detail-page-content/TVMRVVObject';
import { VOB } from './detail-page-content/VOB';
import { Woonvergunningen } from './detail-page-content/Woonvergunningen';
import { ZwaarVerkeer } from './detail-page-content/ZwaarVerkeer';
import { useVergunningenDetailData } from './useVergunningenDetailData.hook';
import { useVergunningenThemaData } from './useVergunningenThemaData.hook';
import {
  DecosVergunning,
  VergunningFrontend,
} from '../../../server/services/vergunningen/config-and-types';
import { CaseTypeV2 } from '../../../universal/types/decos-zaken';
import { Datalist } from '../../components/Datalist/Datalist';
import ThemaDetailPagina from '../ThemaPagina/ThemaDetailPagina';
import { WVOSContent } from './detail-page-content/WVOS';
import DocumentListV2 from '../../components/DocumentList/DocumentListV2';
import { PageContentCell } from '../../components/Page/Page';

interface DetailPageContentProps<V> {
  vergunning: V;
}

// TODO: Implement detailpages per case
function DetailPageContent<V extends VergunningFrontend<DecosVergunning>>({
  vergunning,
}: DetailPageContentProps<V>) {
  return (
    <PageContentCell>
      {(function VergunningDetailContent() {
        switch (vergunning.caseType) {
          case CaseTypeV2.TVMRVVObject:
            return <TVMRVVObject vergunning={vergunning} />;
          case CaseTypeV2.EvenementMelding:
            return <EvenementMelding vergunning={vergunning} />;
          case CaseTypeV2.EvenementVergunning:
            return <EvenementVergunning vergunning={vergunning} />;
          case CaseTypeV2.Omzettingsvergunning:
            return <Omzettingsvergunning vergunning={vergunning} />;
          case CaseTypeV2.ERVV:
            return <ERVV vergunning={vergunning} />;
          case CaseTypeV2.Flyeren:
            return <Flyeren vergunning={vergunning} />;
          case CaseTypeV2.AanbiedenDiensten:
            return <AanbiedenDienstenContent vergunning={vergunning} />;
          case CaseTypeV2.NachtwerkOntheffing:
            return <Nachtwerkontheffing vergunning={vergunning} />;
          case CaseTypeV2.ZwaarVerkeer:
            return <ZwaarVerkeer vergunning={vergunning} />;
          case CaseTypeV2.Samenvoegingsvergunning:
          case CaseTypeV2.Onttrekkingsvergunning:
          case CaseTypeV2.OnttrekkingsvergunningSloop:
          case CaseTypeV2.VormenVanWoonruimte:
          case CaseTypeV2.Splitsingsvergunning:
            return <Woonvergunningen vergunning={vergunning} />;
          case CaseTypeV2.VOB:
            return <VOB vergunning={vergunning} />;
          case CaseTypeV2.RVVHeleStad:
            return <RvvHeleStad vergunning={vergunning} />;
          case CaseTypeV2.RVVSloterweg:
            return <RvvSloterweg vergunning={vergunning} />;
          case CaseTypeV2.WVOS:
            return <WVOSContent vergunning={vergunning} />;

          default:
            return (
              <Datalist
                rows={Object.entries(vergunning).map(([label, content]) => ({
                  label,
                  content: JSON.stringify(content),
                }))}
              />
            );
        }
      })()}
    </PageContentCell>
  );
}

export function VergunningDetailPagina() {
  const { vergunningen, isLoading, isError, routes } =
    useVergunningenThemaData();
  const { vergunning, title, documents } =
    useVergunningenDetailData(vergunningen);

  return (
    <ThemaDetailPagina<VergunningFrontend<DecosVergunning>>
      title={title}
      zaak={vergunning}
      isError={isError}
      isLoading={isLoading}
      pageContentTop={
        vergunning && (
          <>
            <DetailPageContent vergunning={vergunning} />
            {!!documents.length && (
              <PageContentCell spanWide={8}>
                <Datalist
                  rows={[
                    {
                      label: 'Documenten',
                      content: (
                        <DocumentListV2
                          documents={documents}
                          columns={['', '']}
                          className="ams-mb--sm"
                        />
                      ),
                    },
                  ]}
                />
              </PageContentCell>
            )}
          </>
        )
      }
      backLink={routes.themePage}
    />
  );
}
