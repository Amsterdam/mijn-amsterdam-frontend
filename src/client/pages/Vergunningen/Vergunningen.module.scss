.SuppressedParagraph {
  --ams-paragraph-color: #{var(--ams-color-neutral-grey3)};
  text-align: center;
}

.VergunningenTableThemaPagina {
  tr {

    th,
    td {
      @media screen and (min-width: $ams-breakpoint-medium) {
        &:nth-child(1) {
          width: 20%;
        }

        &:nth-child(2) {
          width: 60%;
        }

        &:nth-child(3) {
          width: 20%;
        }
      }
    }
  }
}