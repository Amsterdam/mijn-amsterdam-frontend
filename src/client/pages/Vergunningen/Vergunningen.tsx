import { Paragraph } from '@amsterdam/design-system-react';

import { useVergunningenThemaData } from './useVergunningenThemaData.hook';
import {
  DecosVergunning,
  VergunningFrontend,
} from '../../../server/services/vergunningen/config-and-types';
import { PageContentCell } from '../../components/Page/Page';
import ThemaPagina from '../ThemaPagina/ThemaPagina';
import ThemaPaginaTable from '../ThemaPagina/ThemaPaginaTable';

const pageContentTop = (
  <PageContentCell spanWide={8}>
    <Paragraph>
      Hier ziet u een overzicht van uw aanvragen voor vergunningen en
      ontheffingen bij gemeente Amsterdam.
    </Paragraph>
  </PageContentCell>
);

export function VergunningenThemaPagina() {
  const {
    vergunningen,
    isLoading,
    isError,
    tableConfig,
    linkListItems,
    title,
  } = useVergunningenThemaData();

  const tables = Object.entries(tableConfig).map(
    ([
      kind,
      { title, displayProps, filter, sort, listPageRoute, className },
    ]) => {
      return (
        <ThemaPaginaTable<VergunningFrontend<DecosVergunning>>
          key={kind}
          title={title}
          zaken={vergunningen.filter(filter).sort(sort)}
          listPageRoute={listPageRoute}
          displayProps={displayProps}
          className={className}
        />
      );
    }
  );

  return (
    <ThemaPagina
      title={title}
      pageContentTop={pageContentTop}
      linkListItems={linkListItems}
      pageContentMain={tables}
      isError={isError}
      isLoading={isLoading}
    />
  );
}
