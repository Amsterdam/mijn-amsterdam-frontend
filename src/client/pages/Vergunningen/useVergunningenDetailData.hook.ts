import { useParams } from 'react-router';

import { useVergunningDocumentList } from './detail-page-content/useVergunningDocumentsList.hook';
import { DecosZaakBase } from '../../../server/services/decos/config-and-types';
import { VergunningFrontend } from '../../../server/services/vergunningen/config-and-types';

export function useVergunningenDetailData<T extends DecosZaakBase>(
  vergunningen: VergunningFrontend<T>[]
) {
  const { id } = useParams<{ id: VergunningFrontend['id'] }>();
  const vergunning = vergunningen.find((vergunning) => vergunning.id === id);
  const fetchDocumentsUrl = vergunning?.fetchDocumentsUrl;
  const { documents, isError, isLoading } =
    useVergunningDocumentList(fetchDocumentsUrl);

  return {
    vergunning,
    isErrorDocuments: isError,
    isLoadingDocuments: isLoading,
    documents,
    title: vergunning?.title ?? 'Vergunning',
  };
}
