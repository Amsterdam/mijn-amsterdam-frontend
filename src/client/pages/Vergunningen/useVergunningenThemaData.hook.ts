import { isError } from 'lodash';

import {
  linkListItems,
  routes,
  tableConfig,
} from './Vergunningen-thema-config';
import {
  DecosVergunning,
  VergunningFrontend,
} from '../../../server/services/vergunningen/config-and-types';
import { ThemaIDs } from '../../../universal/config/thema';
import { isLoading } from '../../../universal/helpers/api';
import { addLinkElementToProperty } from '../../components/Table/TableV2';
import { ThemaTitles } from '../../config/thema';
import { useAppStateGetter } from '../../hooks/useAppState';
import { useThemaBreadcrumbs } from '../../hooks/useThemaMenuItems';

export function useVergunningenThemaData() {
  const { VERGUNNINGEN } = useAppStateGetter();
  const vergunningen = addLinkElementToProperty<
    VergunningFrontend<DecosVergunning>
  >(VERGUNNINGEN.content ?? [], 'identifier', true);
  const breadcrumbs = useThemaBreadcrumbs(ThemaIDs.VERGUNNINGEN);

  return {
    title: ThemaTitles.VERGUNNINGEN,
    vergunningen,
    isLoading: isLoading(VERGUNNINGEN),
    isError: isError(VERGUNNINGEN),
    tableConfig,
    linkListItems,
    routes,
    breadcrumbs,
  };
}
