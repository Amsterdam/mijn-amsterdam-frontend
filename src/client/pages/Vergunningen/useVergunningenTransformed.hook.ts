import { useMemo } from 'react';

import { Vergunning } from '../../../server/services';
import { defaultDateFormat } from '../../../universal/helpers/date';
import { getCustomTitleForVergunningWithLicensePlates } from '../../../universal/helpers/vergunningen';
import { AppState } from '../../../universal/types';
import { CaseType } from '../../../universal/types/vergunningen';
import { addTitleLinkComponent } from '../../components/Table/Table';

const titleTransformMap: Record<string, unknown> = {
  [CaseType.TouringcarJaarontheffing]:
    getCustomTitleForVergunningWithLicensePlates,
  [CaseType.TouringcarDagontheffing]:
    getCustomTitleForVergunningWithLicensePlates,
  [CaseType.EigenParkeerplaats]: getCustomTitleForVergunningWithLicensePlates,
  [CaseType.EigenParkeerplaatsOpheffen]:
    getCustomTitleForVergunningWithLicensePlates,
};

export function useVergunningenTransformed(
  VERGUNNINGEN: AppState['VERGUNNINGEN']
) {
  const vergunningen: Vergunning[] = useMemo(() => {
    if (!VERGUNNINGEN.content?.length) {
      return [];
    }
    const items: Vergunning[] = VERGUNNINGEN.content.map((item) => {
      const transformer = titleTransformMap[item.caseType];
      return {
        ...item,
        title:
          typeof transformer === 'function' ? transformer(item) : item.title,
        dateRequest: defaultDateFormat(item.dateRequest),
      };
    });
    return items;
  }, [VERGUNNINGEN.content]);

  return vergunningen;
}
