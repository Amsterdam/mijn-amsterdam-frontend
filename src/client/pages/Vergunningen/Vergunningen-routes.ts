import { VergunningDetailPagina } from './VergunningDetail';
import { VergunningenThemaPagina } from './Vergunningen';
import { VergunningenList } from './VergunningenList';

export const VergunningenRoutes = [
  { route: '/vergunningen/lijst/:kind/:page?', Component: VergunningenList },
  { route: '/vergunningen/:caseType/:id', Component: VergunningDetailPagina },
  { route: '/vergunningen', Component: VergunningenThemaPagina },
];
