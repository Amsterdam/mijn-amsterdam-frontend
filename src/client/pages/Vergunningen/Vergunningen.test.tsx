import { render } from '@testing-library/react';

import { generatePath } from 'react-router-dom';
import { MutableSnapshot } from 'recoil';
import vergunningenData from '../../../server/mock-data/json/vergunningen.json';
import { transformVergunningenData } from '../../../server/services/vergunningen';
import { AppRoutes } from '../../../universal/config';
import { appStateAtom } from '../../hooks/useAppState';
import MockApp from '../MockApp';
import Vergunningen from './Vergunningen';

const testState: any = {
  VERGUNNINGEN: {
    status: 'OK',
    content: transformVergunningenData(vergunningenData as any),
  },
};

function initializeState(snapshot: MutableSnapshot) {
  snapshot.set(appStateAtom, testState);
}

describe('<Vergunningen />', () => {
  const routeEntry = generatePath(AppRoutes.VERGUNNINGEN);
  const routePath = AppRoutes.VERGUNNINGEN;

  const Component = () => (
    <MockApp
      routeEntry={routeEntry}
      routePath={routePath}
      component={Vergunningen}
      initializeState={initializeState}
    />
  );

  it('Matches the Full Page snapshot', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});
