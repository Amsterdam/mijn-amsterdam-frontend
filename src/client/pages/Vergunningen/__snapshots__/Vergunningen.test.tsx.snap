// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Vergunningen /> Matches the Full Page snapshot 1`] = `
<DocumentFragment>
  <main
    class="Page OverviewPage Vergunningen"
  >
    <header
      class="PageHeading hasIcon hasBackLink"
    >
      <span
        aria-hidden="true"
        class="Icon"
      >
        <svg
          aria-label="Vergunningen"
          class="ChapterIcon"
          fill="black"
        >
          vergunningen.svg
        </svg>
      </span>
      <div
        class="HeadingInner"
      >
        <a
          class="Button Button__plain Button__lean Linkd BackLink"
          href="/"
        >
          <svg
            aria-hidden="true"
            class="Icon Icon__left"
            fill="#000"
            height="14px"
            width="14px"
          >
            Chevron-Left.svg
          </svg>
          Home
        </a>
        <h2
          class="Heading Heading__large"
        >
          Vergunningen
        </h2>
      </div>
    </header>
    <div
      class="PageContent"
    >
      <p>
        Hier ziet u een overzicht van uw aanvragen voor vergunningen en ontheffingen bij gemeente Amsterdam.
      </p>
      <p>
        <a
          class="Button Button__plain Button__lean Linkd"
          href="https://www.amsterdam.nl/veelgevraagd/?productid=%7BE4341B52-1FC0-4E17-AB68-2B3AFE15160A%7D"
          rel="external noopener noreferrer"
        >
          <svg
            aria-hidden="true"
            class="Icon Icon__left"
            fill="#000"
            height="14px"
            width="14px"
          >
            Chevron-Right.svg
          </svg>
          Ontheffing RVV en TVM aanvragen
        </a>
      </p>
    </div>
    <section
      class="SectionCollapsible SectionCollapsibleCurrent"
    >
      <h3
        class="Heading Heading__mediumLarge Title TitleWithItems"
      >
        <button
          aria-expanded="true"
          class="TitleToggle"
        >
          <svg
            aria-hidden="true"
            class="CaretIcon"
          >
            Chevron-Right.svg
          </svg>
           Lopende aanvragen
        </button>
      </h3>
      <div
        aria-hidden="false"
        class="Panel"
        style="height: 0px;"
      >
        <div
          class="PanelInner"
        >
          <table
            class="Table Table"
          >
            <thead>
              <tr
                class="TableRow"
              >
                <th
                  class="DisplayProp"
                >
                  Kenmerk
                </th>
                <th
                  class="DisplayProp"
                >
                  Soort vergunning
                </th>
                <th
                  class="DisplayProp"
                >
                  Aangevraagd
                </th>
              </tr>
            </thead>
            <tbody>
              <tr
                class="TableRow"
              >
                <td
                  class="DisplayPropTitle"
                >
                  <a
                    class="Button Button__plain Button__lean Linkd"
                    href="/vergunningen/detail/1862294070"
                  >
                    <svg
                      aria-hidden="true"
                      class="Icon Icon__left"
                      fill="#000"
                      height="14px"
                      width="14px"
                    >
                      Chevron-Right.svg
                    </svg>
                    Z/20/1597197
                  </a>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Soort vergunning:
                  </span>
                  <span>
                    TVM - RVV - Object
                  </span>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Aangevraagd:
                  </span>
                  <span>
                    24 juni 2020
                  </span>
                </td>
              </tr>
              <tr
                class="TableRow"
              >
                <td
                  class="DisplayPropTitle"
                >
                  <a
                    class="Button Button__plain Button__lean Linkd"
                    href="/vergunningen/detail/881727095"
                  >
                    <svg
                      aria-hidden="true"
                      class="Icon Icon__left"
                      fill="#000"
                      height="14px"
                      width="14px"
                    >
                      Chevron-Right.svg
                    </svg>
                    Z/20/1597196
                  </a>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Soort vergunning:
                  </span>
                  <span>
                    TVM - RVV - Object
                  </span>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Aangevraagd:
                  </span>
                  <span>
                    24 juni 2020
                  </span>
                </td>
              </tr>
              <tr
                class="TableRow"
              >
                <td
                  class="DisplayPropTitle"
                >
                  <a
                    class="Button Button__plain Button__lean Linkd"
                    href="/vergunningen/detail/1619362356"
                  >
                    <svg
                      aria-hidden="true"
                      class="Icon Icon__left"
                      fill="#000"
                      height="14px"
                      width="14px"
                    >
                      Chevron-Right.svg
                    </svg>
                    Z/20/1597195
                  </a>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Soort vergunning:
                  </span>
                  <span>
                    TVM - RVV - Object
                  </span>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Aangevraagd:
                  </span>
                  <span>
                    24 juni 2020
                  </span>
                </td>
              </tr>
              <tr
                class="TableRow"
              >
                <td
                  class="DisplayPropTitle"
                >
                  <a
                    class="Button Button__plain Button__lean Linkd"
                    href="/vergunningen/detail/638795381"
                  >
                    <svg
                      aria-hidden="true"
                      class="Icon Icon__left"
                      fill="#000"
                      height="14px"
                      width="14px"
                    >
                      Chevron-Right.svg
                    </svg>
                    Z/20/1597194
                  </a>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Soort vergunning:
                  </span>
                  <span>
                    TVM - RVV - Object
                  </span>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Aangevraagd:
                  </span>
                  <span>
                    24 juni 2020
                  </span>
                </td>
              </tr>
              <tr
                class="TableRow"
              >
                <td
                  class="DisplayPropTitle"
                >
                  <a
                    class="Button Button__plain Button__lean Linkd"
                    href="/vergunningen/detail/1608378930"
                  >
                    <svg
                      aria-hidden="true"
                      class="Icon Icon__left"
                      fill="#000"
                      height="14px"
                      width="14px"
                    >
                      Chevron-Right.svg
                    </svg>
                    Z/20/1597193
                  </a>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Soort vergunning:
                  </span>
                  <span>
                    TVM - RVV - Object
                  </span>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Aangevraagd:
                  </span>
                  <span>
                    24 juni 2020
                  </span>
                </td>
              </tr>
              <tr
                class="TableRow"
              >
                <td
                  class="DisplayPropTitle"
                >
                  <a
                    class="Button Button__plain Button__lean Linkd"
                    href="/vergunningen/detail/4043653692"
                  >
                    <svg
                      aria-hidden="true"
                      class="Icon Icon__left"
                      fill="#000"
                      height="14px"
                      width="14px"
                    >
                      Chevron-Right.svg
                    </svg>
                    Z/20/1592761
                  </a>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Soort vergunning:
                  </span>
                  <span>
                    TVM - RVV - Object
                  </span>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Aangevraagd:
                  </span>
                  <span>
                    08 juni 2020
                  </span>
                </td>
              </tr>
              <tr
                class="TableRow"
              >
                <td
                  class="DisplayPropTitle"
                >
                  <a
                    class="Button Button__plain Button__lean Linkd"
                    href="/vergunningen/detail/211045982"
                  >
                    <svg
                      aria-hidden="true"
                      class="Icon Icon__left"
                      fill="#000"
                      height="14px"
                      width="14px"
                    >
                      Chevron-Right.svg
                    </svg>
                    Z/20/1592552
                  </a>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Soort vergunning:
                  </span>
                  <span>
                    TVM - RVV - Object
                  </span>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Aangevraagd:
                  </span>
                  <span>
                    06 mei 2020
                  </span>
                </td>
              </tr>
              <tr
                class="TableRow"
              >
                <td
                  class="DisplayPropTitle"
                >
                  <a
                    class="Button Button__plain Button__lean Linkd"
                    href="/vergunningen/detail/2226696250"
                  >
                    <svg
                      aria-hidden="true"
                      class="Icon Icon__left"
                      fill="#000"
                      height="14px"
                      width="14px"
                    >
                      Chevron-Right.svg
                    </svg>
                    Z/20/1592521
                  </a>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Soort vergunning:
                  </span>
                  <span>
                    TVM - RVV - Object
                  </span>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Aangevraagd:
                  </span>
                  <span>
                    03 mei 2020
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </section>
    <section
      class="SectionCollapsible SectionCollapsiblePrevious"
    >
      <h3
        class="Heading Heading__mediumLarge Title TitleWithItems"
      >
        <button
          aria-expanded="false"
          class="TitleToggle"
        >
          <svg
            aria-hidden="true"
            class="CaretIcon"
          >
            Chevron-Right.svg
          </svg>
           Eerdere aanvragen
        </button>
      </h3>
      <div
        aria-hidden="true"
        class="Panel"
        style="height: 0px;"
      >
        <div
          class="PanelInner"
        >
          <table
            class="Table Table"
          >
            <thead>
              <tr
                class="TableRow"
              >
                <th
                  class="DisplayProp"
                >
                  Kenmerk
                </th>
                <th
                  class="DisplayProp"
                >
                  Soort vergunning
                </th>
                <th
                  class="DisplayProp"
                >
                  Resultaat
                </th>
              </tr>
            </thead>
            <tbody>
              <tr
                class="TableRow"
              >
                <td
                  class="DisplayPropTitle"
                >
                  <a
                    class="Button Button__plain Button__lean Linkd"
                    href="/vergunningen/detail/1515509663"
                  >
                    <svg
                      aria-hidden="true"
                      class="Icon Icon__left"
                      fill="#000"
                      height="14px"
                      width="14px"
                    >
                      Chevron-Right.svg
                    </svg>
                    Z/20/1597204
                  </a>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Soort vergunning:
                  </span>
                  <span>
                    TVM - RVV - Object
                  </span>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Resultaat:
                  </span>
                  <span>
                    Verleend
                  </span>
                </td>
              </tr>
              <tr
                class="TableRow"
              >
                <td
                  class="DisplayPropTitle"
                >
                  <a
                    class="Button Button__plain Button__lean Linkd"
                    href="/vergunningen/detail/3607311896"
                  >
                    <svg
                      aria-hidden="true"
                      class="Icon Icon__left"
                      fill="#000"
                      height="14px"
                      width="14px"
                    >
                      Chevron-Right.svg
                    </svg>
                    Z/20/1597203
                  </a>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Soort vergunning:
                  </span>
                  <span>
                    TVM - RVV - Object
                  </span>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Resultaat:
                  </span>
                  <span>
                    Ingetrokken
                  </span>
                </td>
              </tr>
              <tr
                class="TableRow"
              >
                <td
                  class="DisplayPropTitle"
                >
                  <a
                    class="Button Button__plain Button__lean Linkd"
                    href="/vergunningen/detail/926876057"
                  >
                    <svg
                      aria-hidden="true"
                      class="Icon Icon__left"
                      fill="#000"
                      height="14px"
                      width="14px"
                    >
                      Chevron-Right.svg
                    </svg>
                    Z/20/1597202
                  </a>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Soort vergunning:
                  </span>
                  <span>
                    TVM - RVV - Object
                  </span>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Resultaat:
                  </span>
                  <span>
                    Niet verleend
                  </span>
                </td>
              </tr>
              <tr
                class="TableRow"
              >
                <td
                  class="DisplayPropTitle"
                >
                  <a
                    class="Button Button__plain Button__lean Linkd"
                    href="/vergunningen/detail/3240785464"
                  >
                    <svg
                      aria-hidden="true"
                      class="Icon Icon__left"
                      fill="#000"
                      height="14px"
                      width="14px"
                    >
                      Chevron-Right.svg
                    </svg>
                    Z/20/1597199
                  </a>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Soort vergunning:
                  </span>
                  <span>
                    TVM - RVV - Object
                  </span>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Resultaat:
                  </span>
                  <span>
                    Verleend
                  </span>
                </td>
              </tr>
              <tr
                class="TableRow"
              >
                <td
                  class="DisplayPropTitle"
                >
                  <a
                    class="Button Button__plain Button__lean Linkd"
                    href="/vergunningen/detail/2952768564"
                  >
                    <svg
                      aria-hidden="true"
                      class="Icon Icon__left"
                      fill="#000"
                      height="14px"
                      width="14px"
                    >
                      Chevron-Right.svg
                    </svg>
                    Z/20/1595179
                  </a>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Soort vergunning:
                  </span>
                  <span>
                    TVM - RVV - Object
                  </span>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Resultaat:
                  </span>
                  <span>
                    Verleend
                  </span>
                </td>
              </tr>
              <tr
                class="TableRow"
              >
                <td
                  class="DisplayPropTitle"
                >
                  <a
                    class="Button Button__plain Button__lean Linkd"
                    href="/vergunningen/detail/2618219588"
                  >
                    <svg
                      aria-hidden="true"
                      class="Icon Icon__left"
                      fill="#000"
                      height="14px"
                      width="14px"
                    >
                      Chevron-Right.svg
                    </svg>
                    Z/20/615500
                  </a>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Soort vergunning:
                  </span>
                  <span>
                    TVM - RVV - Object
                  </span>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Resultaat:
                  </span>
                  <span>
                    Verleend
                  </span>
                </td>
              </tr>
              <tr
                class="TableRow"
              >
                <td
                  class="DisplayPropTitle"
                >
                  <a
                    class="Button Button__plain Button__lean Linkd"
                    href="/vergunningen/detail/4195945502"
                  >
                    <svg
                      aria-hidden="true"
                      class="Icon Icon__left"
                      fill="#000"
                      height="14px"
                      width="14px"
                    >
                      Chevron-Right.svg
                    </svg>
                    Z/20/1597205
                  </a>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Soort vergunning:
                  </span>
                  <span>
                    TVM - RVV - Object
                  </span>
                </td>
                <td
                  class="DisplayProp"
                >
                  <span
                    class="DisplayPropLabel"
                  >
                    Resultaat:
                  </span>
                  <span>
                    Verleend
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </section>
  </main>
</DocumentFragment>
`;
